src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -116,15 +116,7 @@ public ITagHelperBoundAttributeDescriptorBuilder AddDiagnostic(RazorDiagnostic d
 
         public BoundAttributeDescriptor Build()
         {
-            var validationDiagnostics = Validate();
-
-            var diagnostics = new List<RazorDiagnostic>();
-            diagnostics.AddRange(validationDiagnostics);
-
-            if (_diagnostics != null)
-            {
-                diagnostics.AddRange(_diagnostics);
-            }
+            Validate();
```

Comment:
HashSet helps fight this though. How is it different than storing the `RazorDiagnostic`'s that are manually added and not adding a way to mutate that list?